### DISCUSSION TITLE
Incorrect Path to tool.js

### DISCUSSION THREAD

**Discussion by atmediauk** [nova-kit/nova-packages-tool#12] [Q&A] [answered]
Hello,

Since yesterday (14 Sept) ```Nova::allScripts()``` seems to return the wrong path for ```nova-kit``` in my console command.

e.g:

```php artisan combine:nova-tools-js```

```
[webpack-cli] [Error: ENOENT: no such file or directory, open '/vendor/nova-kit/nova-packages-tool/tool.js'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/vendor/nova-kit/nova-packages-tool/tool.js'
}
```
I think it might be missing dist?

**Comment by crynobone** [comment, marked as answer]
It seem that the artisan command never considers remote script, which is possible via `Nova::remoteScript()`.

**Reply by atmediauk** [reply]
> It seem that the artisan command never considers remote script, which is possible via `Nova::remoteScript()`.

Sorry if I'm misunderstanding @crynobone, I thought ```Nova::remoteScript()``` is used to register a new remote asset? In my case I am just trying to retrieve a list of all the existing scripts.

For example the command is:

```
    public function handle()
    {
        Auth::login(User::where('role', '=', 'Admin')->first(), true); // user with full access to nova

        $url = config('app.url') . '/dashboard';
        app()->handle(Request::create($url)); // or whatever your nova root route is

        // dd(Nova::allScripts());

        foreach (Nova::allScripts() as $file) {
            $this->info($file->path());
        } 
    }
```

This outputs the following (pasted partial):

```
/vendor/nova-kit/nova-packages-tool/tool.js
/Users/foo/Sites/test/vendor/coroowicaksono/chart-js-integration/src/../dist/js/chart-js-integration.js
/Users/foo/Sites/test/vendor/dillingham/nova-attach-many/src/Providers/../../dist/js/field.js
/Users/foo/Sites/test/vendor/dillingham/nova-button/src/Providers/../../dist/js/field.js
/Users/foo/Sites/test/vendor/ebess/advanced-nova-media-library/src/../dist/js/field.js
...
...
```
The two things I noticed were:
1. The path for ```nova-kit``` is relative (I'm not sure if this needs to be absolute)
2. The path should point to vendor/nova-kit/nova-packages-tool/**dist**/tool.js and not /vendor/nova-kit/nova-packages-tool/tool.js

This impacts my webpack file:
```
var novaJs = [];

child_process.exec('php artisan combine:nova-tools-js', {cwd: './'}, function(error, stdout, stderr){
    if (error) throw error;

    // continue and convert string output to array
    novaJs = stdout.split(/[\r\n|\n|\r]/).filter(String);

    console.log(novaJs);
    mix.scripts(novaJs, 'public/js/nova-tools.js').vue();

}); 
```
because when I run ```npm run production``` it fails due to an incorrect path for ```nova-kit```.

```
[webpack-cli] [Error: ENOENT: no such file or directory, open '/vendor/nova-kit/nova-packages-tool/tool.js'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/vendor/nova-kit/nova-packages-tool/tool.js'
}
```
Hopefully that makes more sense with what I'm trying to achieve.

Thanks again.

**Reply by crynobone** [reply]
`/vendor/nova-kit/nova-packages-tool/tool.js`

This is correct because it is served from the public path.

```html
<script src="/vendor/nova-kit/nova-packages-tool/tool.js" />
```

**Reply by atmediauk** [reply]
Ah sorry I didn't realise it was served from the public path 🤦‍♂️. Thank you.

I added a remote check and formatted those paths accordingly:

```
        foreach (Nova::allScripts() as $file) {

            if ($file->isRemote()) {
                $this->info(public_path($file->path()));
            } else {
                $this->info($file->path());         
            }
        } 
```

Thanks again @crynobone
